python/baseline/tf/classify/model.py

```diff
@@ -281,62 +269,48 @@ def load(cls, basename, **kwargs):
         
         :return: A restored model
         """
-        sess = kwargs.get('session', kwargs.get('sess', tf.Session()))
-        model = cls()
-        with open(basename + '.saver') as fsv:
-            saver_def = tf.train.SaverDef()
-            text_format.Merge(fsv.read(), saver_def)
-
-        checkpoint_name = kwargs.get('checkpoint_name', basename)
-        checkpoint_name = checkpoint_name or basename
-
         state = read_json(basename + '.state')
-
-        for prop in ls_props(model):
-            if prop in state:
-                setattr(model, prop, state[prop])
-
-        with gfile.FastGFile(basename + '.graph', 'rb') as f:
-            gd = tf.GraphDef()
-            gd.ParseFromString(f.read())
-            sess.graph.as_default()
-            tf.import_graph_def(gd, name='')
-            try:
-                sess.run(saver_def.restore_op_name, {saver_def.filename_tensor_name: checkpoint_name})
-            except:
-                # Backwards compat
-                sess.run(saver_def.restore_op_name, {saver_def.filename_tensor_name: checkpoint_name + ".model"})
-
-        model.embeddings = dict()
-        for key, class_name in state['embeddings'].items():
+        state['sess'] = kwargs.pop('sess', tf.Session())
+        state['model_type'] = kwargs.get('model_type', 'default')
+        embeddings = dict()
+        embeddings_dict = state.pop('embeddings')
+        for key, class_name in embeddings_dict.items():
             md = read_json('{}-{}-md.json'.format(basename, key))
             embed_args = dict({'vsz': md['vsz'], 'dsz': md['dsz']})
```

Comment:
Are embedding objects also getting this save kwargs treatment? If so shouldn't the embed args be the whole contents of the md file?